app/src/main/java/com/andrew/tasky/auth/presentation/screens/register/RegisterViewModel.kt

```diff
@@ -22,18 +25,28 @@ class RegisterViewModel @Inject constructor(
 
     private val _isNameValid = MutableStateFlow(false)
     val isNameValid = _isNameValid.asStateFlow()
-    fun setIsNameValid(nameValidity: Boolean) {
-        _isNameValid.value = nameValidity
+    fun setIsNameValid(name: String) {
+        _isNameValid.value = NameValidator().validate(name)
+    }
+
+    fun isPasswordValid(password: String): Boolean {
+        return PasswordValidator().validate(password)
     }
 
     fun register(name: String, email: String, password: String) {
-        viewModelScope.launch {
-            val result = repository.register(
-                name = name,
-                email = email,
-                password = password
-            )
-            resultChannel.send(result)
+        if (
+            isPasswordValid(password) &&
+            isNameValid.value &&
+            Patterns.EMAIL_ADDRESS.matcher(email).matches()
```

Comment:
`Patterns` is a class from the Android SDK, which means this ViewModel won't be locally unit testable. To avoid this, you can do this:
*In domain*
```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

*In data*
```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an instance of `EmailPatternValidator` in your ViewModel, so it's not forced to use your specific implementation. In local unit tests you could then provide a fake version of it that doesn't use the underlying Android-dependency.